include/zephyr/sys/min_heap.h

```diff
@@ -0,0 +1,247 @@
+ * upward as necessary. Insert operation will fail if the min-heap
+ * has reached full capacity.
+ *
+ * @param heap Pointer to the min-heap.
+ * @param item Pointer to the item to insert.
+ *
+ * @return 0 on Success, -ENOMEM if the heap is full.
+ */
+int min_heap_push(struct min_heap *heap, const void *item);
+
+/**
+ * @brief Peek at the top element of the min-heap.
+ *
+ * The function will not remove the element from the min-heap.
+ *
+ * @param heap Pointer to the min-heap.
+ *
+ * @return Pointer to the top priority element, or NULL if the heap is empty.
+ */
+void *min_heap_peek(const struct min_heap *heap);
+
+/**
+ * @brief Remove a specific element from the min-heap.
+ *
+ * Removes the specified node from the min-heap based on the ID it stores
+ * internally. The min-heap is rebalanced after removal to ensure
+ * proper ordering.
+ * The caller gains ownership of the returned element and is responsible for
+ * any further management of its memory or reuse.
+ *
+ * @param heap Pointer to the min-heap.
+ * @param id element ID to be removed.
+ * @param out_buf User-provided buffer where the removed element will be copied.
+ *
+ * @return true in success, false otherwise.
+ */
+bool min_heap_remove(struct min_heap *heap, size_t id, void *out_buf);
+
+/**
+ * @brief Check if the min heap is empty.
+ *
+ * This function checks whether the heap contains any elements.
+ *
+ * @param heap Pointer to the min heap.
+ *
+ * @return true if heap is empty, false otherwise.
+ */
+static inline bool min_heap_is_empty(struct min_heap *heap)
+{
+	__ASSERT_NO_MSG(heap != NULL);
+
+	return (heap->size == 0);
+}
+
+/**
+ * @brief Remove and return the highest priority element in the heap.
+ *
+ * The caller gains ownership of the returned element and is responsible for
+ * any further management of its memory or reuse. The min-heap is rebalanced
+ * after removal to ensure proper ordering.
+ *
+ * @param heap Pointer to heap.
+ * @param out_buf User-provided buffer where the removed element will be copied.
+ *
+ * @return true in success, false otherwise.
+ */
+bool min_heap_pop(struct min_heap *heap, void *out_buf);
+
+/**
+ * @brief Search for a node in the heap matching a condition.
+ *
+ * @param heap Pointer to the heap structure.
+ * @param eq Function used to compare each node with the search target.
+ * @param other Pointer to the data used for comparison in the eq function.
+ * @param out_id Optional output parameter to store the index of the found node.
+ *
+ * @return Pointer to the first matching element, or NULL if not found.
+ */
+void *min_heap_find(struct min_heap *heap, min_heap_eq_t eq,
+				     const void *other, size_t *out_id);
+
+/**
+ * @brief Get a pointer to the element at the specified index.
+ *
+ * @param heap Pointer to the min-heap.
+ * @param index Index of the element to retrieve.
+ *
+ * @return Pointer to the element at the given index.
+ */
+static inline void *get_element(const struct min_heap *heap, size_t index)
```

Comment:
This function should be prefixed and / or perhaps in the .c file
```suggestion
static inline void *min_heap_get_element(const struct min_heap *heap, size_t index)
```

Reply:
I will prefix it, as _FOR_EACH macro is using the function to parse the heap.

Reply:
Changed.